File: 01_Language/01_Functions/python/hash_file.py
# coding: utf-8

import hashlib


def hash_file(algo, filename):
    with open(filename, mode="rb") as f:
        h = hashlib.new(algo)
        while True:
            data = f.read(8192)
            if not data:
                break
            h.update(data)
        return h.hexdigest()


if __name__ == '__main__':
    with open("test.txt", mode="w") as f:
        f.write("The quick brown fox jumped over the lazy dog.")
    print(hash_file("md5", "test.txt"))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/HashFile.java
package com.onevgo.functions;

import cn.hutool.crypto.digest.Digester;

import java.io.File;

public class HashFile {
    public static String hashFile(String algo, String filename) {
        return new Digester(algo).digestHex(new File(filename));
    }

    public static void main(String[] args) {
        FilePutContents.filePutContents("test.txt", "The quick brown fox jumped over the lazy dog.");
        System.out.println(hashFile("md5", "test.txt"));
    }
}
